src/ray/core_worker/experimental_mutable_object_manager.cc

```diff
@@ -289,16 +292,27 @@ Status MutableObjectManager::ReadAcquire(const ObjectID &object_id,
   // Check whether the channel has an error set before checking that we are the only
   // reader. If the channel is already closed, then it's OK to ReadAcquire and
   // ReadRelease in any order.
+
+  auto timeout_point = ToTimeoutPoint(timeout_ms);
+  bool locked = false;
+  bool expired = false;
   do {
     RAY_RETURN_NOT_OK(object->header->CheckHasError());
     // The channel is still open. This lock ensures that there is only one reader
     // at a time. The lock is released in `ReadRelease()`.
-  } while (!channel->lock->try_lock());
-  channel->reading = true;
+    locked = channel->lock->try_lock();
+    expired = timeout_point && std::chrono::steady_clock::now() >= *timeout_point;
+  } while (!locked && !expired);
+  if (!locked) {
```

Comment:
This is a bit weird that we check `locked` instead of `expired` for timeout error. It may be more readable if we use `expired`.

Reply:
See the comment right below this line why we need to check `locked` and not `expired`.